File: Problems/429_N-ary_Tree_Level_Order_Traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        if (!root) return res;
        queue<Node*> qu;
        qu.push(root);
        while (!qu.empty()) {
            int len = qu.size();
            vector<int> tmp;
            for (int i = 0; i < len; ++i) {
                Node* curr = qu.front();
                qu.pop();
                tmp.push_back(curr -> val);
                for (const auto &node : curr -> children) {
                    qu.push(node);
                }
            }
            res.push_back(tmp);
        }
        return res;
    }
};

File: Problems/429_N-ary_Tree_Level_Order_Traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        res = []
        if not root:
            return res
        qu = deque([root])
        while qu:
            length = len(qu)
            tmp = []
            for _ in range(length):
                curr = qu.popleft()
                tmp.append(curr.val)
                for child in curr.children:
                    qu.append(child)
            res.append(tmp)
        return res